dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,239 @@
+from __future__ import unicode_literals
+
+import os
+import posixpath
+
+from funcy import cached_property
+from ratelimit import limits, sleep_and_retry
+from backoff import on_exception, expo
+
+from dvc.scheme import Schemes
+from dvc.path_info import CloudURLInfo
+from dvc.remote.base import RemoteBASE
+from dvc.config import Config
+from dvc.exceptions import DvcException
+from dvc.remote.gdrive.pydrive import (
+    RequestListFile,
+    RequestListFilePaginated,
+    RequestUploadFile,
+    RequestDownloadFile,
+)
+from dvc.remote.gdrive.utils import FOLDER_MIME_TYPE
+
+
+class GDriveURLInfo(CloudURLInfo):
+    @property
+    def netloc(self):
+        return self.parsed.netloc
+
+
+class RemoteGDrive(RemoteBASE):
+    scheme = Schemes.GDRIVE
+    path_cls = GDriveURLInfo
+    REGEX = r"^gdrive://.*$"
+    REQUIRES = {"pydrive": "pydrive"}
+
+    def __init__(self, repo, config):
+        super(RemoteGDrive, self).__init__(repo, config)
+        self.no_traverse = False
+        self.cached_dirs = {}
+        self.cached_ids = {}
+        self.path_info = self.path_cls(config[Config.SECTION_REMOTE_URL])
+        self.config = config
+        self.init_drive()
+
+    def init_drive(self):
+        if Config.SECTION_REMOTE_KEY_FILE not in self.config:
+            raise DvcException(
+                "Google Drive settings file path is missed from config. "
+                "Learn more at https://dvc.org/doc."
+            )
+        self.gdrive_credentials_path = self.config.get(
+            Config.SECTION_REMOTE_KEY_FILE
+        )
+        self.root_id = self.get_path_id(self.path_info, create=True)
+        self.cache_root_dirs()
+
+    @on_exception(expo, DvcException, max_tries=8)
+    @sleep_and_retry
+    @limits(calls=10, period=1)
+    def execute_request(self, request):
+        try:
+            result = request.execute()
+        except Exception as exception:
+            if "Rate Limit Exceeded" in str(exception):
```

Comment:
better test code?

Reply:
this looks fragile as it is now

Reply:
Not sure that response code is included in Exception's message. Will check again together with possibility to catch specific PyDrive's exceptions (had issues to catch them all).

Reply:
thanks, changed to look for `HttpError 403` in Exception's string